### PR TITLE
feat: Worked on the Job entity and creating new job listing

### PR DESCRIPTION
**Pull request by Bola-coder**
### PR Description
**What does the PR do?**
This PR is about creating a new job endpoint.

**How should this be manually tested?**
Request Body
POST /api/jobs

`{
  "title": "Senior Developer",
  "description": "Responsible for developing and maintaining software applications.",
  "location": "Remote",
  "salary": "120000",
  "job_type": "Full-time"
}
`

**Successful Response**
On a successful creation of the job, the API should return a 201 Created status code.
The response body should contain the newly created job details.
`{
  "message": "Job created successfully",
  "job": {
    "title": "Senior Developer",
    "description": "Responsible for developing and maintaining software applications.",
    "location": "Remote",
    "salary": "120000",
    "job_type": "Full-time"
  }
}
`

**Authentication Error**
When the current user is not properly authenticated to create a job:

`{
  "status_code": 401,
  "error": "invalid authentication token",
  "message": "user not authorized for this action"
}
`

**Validation**
If any required field is missing or invalid, the response should return a 400 Bad Request status code with appropriate validation error messages:

`{
  "status_code": 400,
  "message": ["title must be a string", "salary must be a string"],
  "error": "Bad Request"
}
`

**Task**
Create a POST endpoint POST /api/v1/jobs/ to create a new job.
Implement server-side logic to handle successful and failed create requests.
Ensure the user has the proper permissions and authentication to create a job.
Write tests for creating job.
Update API documentation to reflect the new endpoint and its usage.

**Reference Issue**
[FEAT:](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/159)

### CODE REVIEW

File: src/database/data-source.ts
Comment (Heba-WebDev) [inline src/database/data-source.ts:1]: Why was this line added? Please remove it

File: src/modules/job/job.controller.ts
Comment (Heba-WebDev) [inline src/modules/job/job.controller.ts:1]: The API doc status should be 201

Comment (Bola-coder) [reply src/modules/job/job.controller.ts:1]: @Heba-WebDev @Homoakin619 

I have corrected it

**Review by Heba-WebDev — APPROVED** [review]

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
Failing ci/cd

**Review by Heba-WebDev — APPROVED** [review]
